src/request-data/user-preferences/normalizers.js

```diff
@@ -33,6 +33,21 @@ export class SitePreferenceNormalizer extends PreferenceNormalizer {
   static projectSortMode(val) {
     return ['alphabetical', 'latest', 'newest'].includes(val) ? val : 'latest';
   }
+
+  static outdatedVersionWarningDismissDate(val) {
+    // Frontend to Backend
+    if (val instanceof Date) {
+      return val.toISOString();
+    }
+
+    // Backend to Frontend
+    const isoDateRegex = /^\d{4}-\d{2}-\d{2}([T](\d{2}:\d{2}:\d{2}(\.\d+)?Z)?)?$/;
```

Comment:
I mentioned `DateTime.fromISO().isValid` in the tests below, but I want to be explicit that I don't think we should use it here. The `luxon` package is kind of heavy, so I want to avoid including it in the main bundle.

I could see it being helpful one day to have a utility function like `isISODate()`.